# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptLoader.h
# Commit: d52858ae58d2
# Full Hash: d52858ae58d25708320093094a8ff791eacf2b15
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-21 21:39:46
# Regressor Bug: 1742437
# File Overlap Count: 1
# Description:
#   Bug 1742437 - Split ScriptLoadRequest into ScriptLoadRequest and DOMScriptLoadContext; r=jonco,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D134041
# ==============================================================================

diff -r 0355b14a5461 -r d52858ae58d2 dom/script/ScriptLoader.h
--- a/dom/script/ScriptLoader.h	Mon Feb 21 14:53:59 2022 +0000
+++ b/dom/script/ScriptLoader.h	Mon Feb 21 15:39:04 2022 +0000
@@ -559,6 +559,12 @@
   // Implements https://html.spec.whatwg.org/#execute-the-script-block
   nsresult EvaluateScriptElement(ScriptLoadRequest* aRequest);
 
+  // Handles both offthread and synchronous modules
+  nsresult CompileOrFinishModuleScript(
+      JSContext* aCx, JS::Handle<JSObject*> aGlobal,
+      JS::CompileOptions& aOptions, ModuleLoadRequest* aRequest,
+      JS::MutableHandle<JSObject*> aModule) override;
+
   // Handles both bytecode and text source scripts; populates exec with a
   // compiled script
   nsresult CompileOrDecodeClassicScript(JSContext* aCx,
